client/src/lib/interface.ts

```diff
@@ -106,3 +129,146 @@ export interface Pin {
 }
 
 export type PinStatus = 'queued' | 'pinning' | 'pinned' | 'failed'
+
+/**
+ * This is an input used to construct the Token metadata as per EIP-1155
+ * @see https://eips.ethereum.org/EIPS/eip-1155#metadata
+ */
+export interface TokenInput {
+  /**
+   * Identifies the asset to which this token represents
+   */
+  name: string
+  /**
+   * Describes the asset to which this token represents
+   */
+  description: string
+  /**
+   * An `File` with mime type image/* representing the asset this
+   * token represents. Consider making any images at a width between `320` and
+   * `1080` pixels and aspect ratio between `1.91:1` and `4:5` inclusive.
+   *
+   * If `File` object is used, URL in the metadata will include a filename
+   * e.g. `ipfs://bafy...hash/cat.png`. If `Blob` is used URL in the metadata
+   * will not include filename or extension e.g. `ipfs://bafy...img/`
+   */
+  image: Blob | File
+
+  /**
+   * The number of decimal places that the token amount should display - e.g.
+   * `18`, means to divide the token amount by `1000000000000000000` to get its
+   * user representation.
+   */
+  decimals?: number
+
+  /**
+   * Arbitrary properties. Values may be strings, numbers, nested object or
+   * arrays of values. It is possible to provide a `File` or a `Blob` instance
+   * as property value, in which case it is stored on IPFS and metadata will
+   * contain URL to it in form of `ipfs://bafy...hash/name.png` or
+   * `ipfs://bafy...file/` respectively.
+   */
+  properties?: Object
```

Comment:
I think we really should at least request it to be an object otherwise all of the nice type inference that took me a while to get will break. And really we should discourage sticking primitives or files in there.